src/fsharp/ConstraintSolver.fs

```diff
@@ -1033,6 +1036,8 @@ and SolveTypeEqualsType (csenv: ConstraintSolverEnv) ndeep m2 (trace: OptionalTr
         SolveTypeEqualsTypeKeepAbbrevs csenv ndeep m2 trace rty1 rty2 
 
     | TType_ucase (uc1, l1), TType_ucase (uc2, l2) when g.unionCaseRefEq uc1 uc2  -> SolveTypeEqualsTypeEqns csenv ndeep m2 trace None l1 l2
+    | TType_erased_union _, TType_erased_union _ when typeAEquiv g aenv sty1 sty2 -> 
```

Comment:
This should allow solution, e.g.

    (?1 list | int) =~= (string list | int) 

should solve inference variable `?1` to `string`

I beleive your `TType_erased_union` is already in canonical flattened order, so per my comment here: https://github.com/fsharp/fslang-design/pull/512/files#r556027949 you can just solve each element to be equivalent?